sklearn/cluster/kernel_kmeans_.py

```diff
@@ -0,0 +1,265 @@
+"""
+    Kernel K-means clustering
+
+    Reference
+    ---------
+    Kernel k-means, Spectral Clustering and Normalized Cuts.
+    Inderjit S. Dhillon, Yuqiang Guan, Brian Kulis.
+    KDD 2004.
+
+    Author: Mathieu Blondel <mathieu@mblondel.org>
+            Ishank Gulati <gulati.ishank@gmail.com>
+
+    License: BSD 3 clause
+"""
+from __future__ import print_function
+from __future__ import division
+
+import numpy as np
+
+from ..base import BaseEstimator, ClusterMixin
+from ..metrics.pairwise import pairwise_kernels
+from ..utils import check_random_state
+from ..externals.six.moves import xrange
+from ..utils import check_array
+from ..utils.validation import FLOAT_DTYPES
+
+
+class KernelKMeans(BaseEstimator, ClusterMixin):
+    """
+    Parameters
+    ----------
+
+    n_clusters : int, default: 3
+        The number of clusters to be formed as well as number of centroids
+        to be generated.
+
+    max_iter : int, default: 300
+        Maximum number of iterations of algorithm for a single run.
+
+    tol : float, default: 1e-3
+        Relative tolerance with regards to inertia to declare convergence.
+
+    random_state : integer or numpy.RandomState, optional
+        The generator used to initialize the centers. If an integer is
+        given, it fixes the seed. Defaults to the global numpy random
+        number generator.
+
+    kernel : {'linear_kernel', 'polynomial_kernel', 'sigmoid_kernel',
+              'rbf_kernel', 'chi2_kernel'}, default: 'linear_kernel'
+        The type of kernel to be used in kernel k means algorithm.
```

Comment:
I don't think this will render well in the HTML docs. After the colon you should specify the type, not the allowed values. See [svm](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/svm/classes.py#L404) for an example of what I mean.